examples/animations/smooth_curves.rs

```diff
@@ -0,0 +1,177 @@
+use bevy::{
+    math::{
+        curves::{Curve, CurveCursor, CurveVariable, TangentControl},
+        interpolation::Interpolation,
+    },
+    prelude::*,
+    render::pipeline::PrimitiveTopology,
+};
+
+struct CurveCursorTag;
+
+struct CurveTargetTag;
+
+#[derive(Default)]
+struct CurveMesh {
+    timer: Timer,
+    curve: CurveVariable<f32>,
+}
+
+fn main() {
+    App::build()
+        .insert_resource(CurveMesh::default())
+        .add_plugins(DefaultPlugins)
+        .add_startup_system(setup.system())
+        .add_system(animate.system())
+        .run();
+}
+
+fn values<T: Copy>(length: usize, default: T) -> Vec<T> {
+    let mut v = vec![];
+    v.resize(length, default);
+    v
+}
+
+fn line(a: [f32; 3], b: [f32; 3]) -> Mesh {
+    let mut mesh = Mesh::new(PrimitiveTopology::LineStrip);
+    mesh.set_attribute(Mesh::ATTRIBUTE_POSITION, vec![a, b]);
+    mesh.set_attribute(Mesh::ATTRIBUTE_NORMAL, values(2, [0.0f32, 0.0, 1.0]));
+    mesh.set_attribute(Mesh::ATTRIBUTE_TANGENT, values(2, [0.0f32, 1.0, 0.0]));
+    mesh.set_attribute(Mesh::ATTRIBUTE_UV_0, values(2, [0.0f32; 2]));
+    mesh
+}
+
+fn setup(
+    mut commands: Commands,
+    mut curve_mesh: ResMut<CurveMesh>,
+    mut meshes: ResMut<Assets<Mesh>>,
+    mut materials: ResMut<Assets<StandardMaterial>>,
+) {
+    // Create curve
+    curve_mesh.curve = CurveVariable::with_tangents_and_mode(
+        vec![0.0, 1.0, 1.3, 1.6, 1.7, 1.8, 1.9, 2.0],
+        vec![3.0, 0.0, 1.0, 0.0, 0.5, 0.0, 0.25, 0.0],
+        TangentControl::Auto,
+        Interpolation::Hermite,
+    )
+    .unwrap();
+    // Create timer
+    curve_mesh.timer = Timer::from_seconds(2.5, true);
+
+    // Create curve mesh
+    const DIVS: usize = 1024;
+    let mut mesh = Mesh::new(PrimitiveTopology::LineStrip);
+    mesh.set_attribute(
+        Mesh::ATTRIBUTE_POSITION,
+        (0..DIVS)
+            .into_iter()
+            .map(|i| {
+                let time = (i as f32 / (DIVS - 1) as f32) * curve_mesh.curve.duration();
+                [time, curve_mesh.curve.sample(time), 0.0]
+            })
+            .collect::<Vec<_>>(),
+    );
+    mesh.set_attribute(Mesh::ATTRIBUTE_NORMAL, values(DIVS, [0.0f32, 0.0, 1.0]));
```

Comment:
I would like comments explaining each of the lines where we're setting mesh attributes; it's hard to tell why those specific values are needed and what the various numbers correspond to.